core/dbt/events/functions.py

```diff
@@ -178,6 +182,7 @@ def create_json_log_line(e: T_Event, msg_fn: Callable[[T_Event], str]) -> str:
             in e.__dataclass_fields__.items()  # type: ignore[attr-defined]
             if type(y._field_type) == _FIELD_BASE
         }
+        values['data']['type'] = 'dataclass_attr'
```

Comment:
hmm I don't think this is quite what we're looking for here. `type` should communicate a schema i.e. what fields and types of values to expect, however this seems to communicate the lack of a schema.

Reply:
I'm not really seeing what the schema here is since it is just any attributes that were passed in.  I know we've defined them in the event classes but each one is different.

Reply:
my guess is that "any attributes that were passed in" is a distributed type instead of a centralized type. The schema is written down, it's just written across the whole codebase not in one place. It exists even if it's annoying to pin down. I can help pin this down in the future. we may have to release with this kind of unstructured for now though.